ckanext/noanonaccess/plugin.py

```diff
@@ -13,6 +13,16 @@ class AuthMiddleware(object):
     def __init__(self, app, app_conf):
         self.app = app
     def __call__(self, environ, start_response):
+        # Get the dcat_access variable from the config object
+        dcat_access = config.get('ckanext.noanonaccess.allow_dcat')
+        # List of extensions to be made accessible if dcat_access is True
+        ext = ['.jsonld','.xml','.ttl','.n3']
+        # List of catalog endpoints                                      
+        catalog_endpoint = config.get('ckanext.dcat.catalog_endpoint')
+        catalog_endpoints = ['/catalog']                                 
+        if catalog_endpoint:                  
+            catalog_endpoint = catalog_endpoint.split('/{_format}')      
+            catalog_endpoints.append(catalog_endpoint[0])
```

Comment:
Think this could be written simplet

```
catalog_endpoints = ['/catalog']  + config.get('ckanext.dcat.catalog_endpoint', '').split('/{_format}')
```

Reply:
The functions such as `startswith` or `endswith` don't work with `lists`, so `tuples` are used.

> startswith first arg must be str, unicode, or tuple, not list

We can use tuples instead of using lists first and then converting them to tuples for `ext` and `catalog_endpoints`

But for that we would have to tweak the above mentioned logic a bit, such as
```
catalog_endpoints = ('/catalog',) + tuple(config.get('ckanext.dcat.catalog_endpoint','').split('/{_format}'))
```
But this gives out `('/catalog', '/dcat/catalog', '')` as `catalog_endpoints` which contains an empty string and enables the home page for anon.

So to fix that, this would work
```
catalog_endpoints = ('/catalog',) + tuple(filter(None,config.get('ckanext.dcat.catalog_endpoint','').split('/{_format}')))
```
Giving out `('/catalog', '/dcat/catalog')`

Reply:
@MuhammadIsmailShahzad ok in that case. Can we move all this inside `elif dcat_access` so that it is more readable? Smth like
```
dcat_access = config.get('ckanext.noanonaccess.allow_dcat')
...
elif dcat_access:
    ext = ['.jsonld','.xml','.ttl','.n3']
        # List of catalog endpoints                                      
        catalog_endpoint = config.get('ckanext.dcat.catalog_endpoint')
        catalog_endpoints = ['/catalog']                                 
        if catalog_endpoint:                  
            catalog_endpoint = catalog_endpoint.split('/{_format}')      
            catalog_endpoints.append(catalog_endpoint[0]) 
       if (environ['PATH_INFO'].endswith(tuple(ext))
                          or environ['PATH_INFO'].startswith(tuple(catalog_endpoints))):
           ...
```